File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/data/daos/IGameStateDao.java
package com.dkellycollins.triggerfinger.data.daos;

import com.dkellycollins.triggerfinger.data.entity.IGameState;

/**
 * Provides the ability to manage game state data.
 */
public interface IGameStateDao extends IActivityDao {

    IGameState retrieve();

    void update(boolean gameOver, boolean isPause);
}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/data/daos/IGameStateDao.kt
package com.dkellycollins.triggerfinger.data.daos

import com.dkellycollins.triggerfinger.data.entity.IGameState

/**
 * Provides the ability to manage game state data.
 */
interface IGameStateDao : IActivityDao {

    fun retrieve(): IGameState

    fun update(gameOver: Boolean, isPause: Boolean)
}
